Lib/weakref.py

```diff
@@ -120,13 +120,17 @@ def remove(wr, selfref=ref(self), _atomic_removal=_remove_dead_weakref):
         self.update(other, **kw)
 
     def _commit_removals(self):
-        l = self._pending_removals
+        pop = self._pending_removals.pop
         d = self.data
         # We shouldn't encounter any KeyError, because this method should
         # always be called *before* mutating the dict.
-        while l:
-            key = l.pop()
-            _remove_dead_weakref(d, key)
+        while True:
+            try:
+                key = pop()
+            except IndexError:
+                return
+            else:
+                _remove_dead_weakref(d, key)
```

Comment:
The reason `self` fields are interned here isn't so much performance but to preserve behavior at interpreter finalization (e.g. if any weakref operation is made as part of `atexit` registrations). This is why there's an alias here:
https://github.com/python/cpython/blob/28db1f61f20352c02e4ae1518e5aeb6505df3045/Lib/weakref.py#L106

I would use the same local alias here too. Other places you touched use built-in `del` or a `discard` that *is* interned.

Reply:
More context: https://bugs.python.org/issue29519